File: backend/app/constants/ensmr.py
"""Structure officielle des départements et filières de l'ENSMR (École Nationale Supérieure des Mines de Rabat)."""

# Dictionnaire : département → liste des filières
ENSMR_DEPARTEMENTS: dict[str, list[str]] = {
    "Département Informatique": [
        "Génie Informatique (GI)",
        "Génie Productique (GP)",
    ],
    "Département Génie des Matériaux": [
        "Matériaux et Contrôle Qualité (MCQ)",
    ],
    "Département Électromécanique": [
        "Électromécanique",
        "Maintenance Industrielle",
        "Efficacité Énergétique et Énergies Renouvelables (3ER)",
    ],
    "Département Génie Industriel": [
        "Management Industriel (MGI)",
        "Management Financier (MGF)",
    ],
    "Département Génie des Procédés Industriels": [
        "Ingénierie des Procédés Industriels (IPI)",
        "Génie Énergétique (GE)",
    ],
    "Département Sciences de la Terre": [
        "GCM - Routes et Ouvrages Hydrauliques",
        "GCM - Génie des Constructions",
        "GCM - Génie Minier",
    ],
    "Département Mines": [
        "Aménagement et Exploitation du Sol et Sous-Sol (AESSS)",
        "Environnement et Sécurité Industriels (ESI)",
    ],
}

# Liste plate des noms de départements
TOUS_LES_DEPARTEMENTS: list[str] = list(ENSMR_DEPARTEMENTS.keys())

# Toutes les filières, toutes sections confondues
TOUTES_LES_FILIERES: list[str] = [
    filiere
    for filieres in ENSMR_DEPARTEMENTS.values()
    for filiere in filieres
]

# Filière → département parent
FILIERE_TO_DEPARTEMENT: dict[str, str] = {
    filiere: dept
    for dept, filieres in ENSMR_DEPARTEMENTS.items()
    for filiere in filieres
}

File: Enim_Connect_Website/src/constants/ensmr.ts
/** Structure officielle des départements et filières de l'ENSMR */

export interface DepartementENSMR {
  nom: string;
  code: string;
  filieres: string[];
}

export const ENSMR_DEPARTEMENTS: DepartementENSMR[] = [
  {
    nom: "Département Informatique",
    code: "INFO",
    filieres: [
      "Génie Informatique (GI)",
      "Génie Productique (GP)",
    ],
  },
  {
    nom: "Département Génie des Matériaux",
    code: "MAT",
    filieres: [
      "Matériaux et Contrôle Qualité (MCQ)",
    ],
  },
  {
    nom: "Département Électromécanique",
    code: "EM",
    filieres: [
      "Électromécanique",
      "Maintenance Industrielle",
      "Efficacité Énergétique et Énergies Renouvelables (3ER)",
    ],
  },
  {
    nom: "Département Génie Industriel",
    code: "GI",
    filieres: [
      "Management Industriel (MGI)",
      "Management Financier (MGF)",
    ],
  },
  {
    nom: "Département Génie des Procédés Industriels",
    code: "GPI",
    filieres: [
      "Ingénierie des Procédés Industriels (IPI)",
      "Génie Énergétique (GE)",
    ],
  },
  {
    nom: "Département Sciences de la Terre",
    code: "ST",
    filieres: [
      "GCM - Routes et Ouvrages Hydrauliques",
      "GCM - Génie des Constructions",
      "GCM - Génie Minier",
    ],
  },
  {
    nom: "Département Mines",
    code: "MINES",
    filieres: [
      "Aménagement et Exploitation du Sol et Sous-Sol (AESSS)",
      "Environnement et Sécurité Industriels (ESI)",
    ],
  },
];

/** Liste plate des noms de départements */
export const NOMS_DEPARTEMENTS: string[] = ENSMR_DEPARTEMENTS.map((d) => d.nom);

/** Toutes les filières */
export const TOUTES_FILIERES: string[] = ENSMR_DEPARTEMENTS.flatMap((d) => d.filieres);

/** Filière → département */
export const FILIERE_TO_DEPARTEMENT: Record<string, string> = Object.fromEntries(
  ENSMR_DEPARTEMENTS.flatMap((d) => d.filieres.map((f) => [f, d.nom]))
);

/** Département → ses filières */
export const DEPT_TO_FILIERES: Record<string, string[]> = Object.fromEntries(
  ENSMR_DEPARTEMENTS.map((d) => [d.nom, d.filieres])
);
